File: tests/fixtures/security/typescript/sql-injection-vulnerable.ts
/*
 * Test Fixture: SQL Injection Vulnerable Patterns (TypeScript)
 *
 * This file contains SQL injection vulnerability patterns for TypeScript that MUST be detected.
 *
 * Expected detections: ≥3 violations
 * Rule ID: SEC-001
 * Severity: CRITICAL
 *
 * NOTE: Stub implementation - full patterns added during Phase 03 (Green)
 */

import { Client } from 'pg';

// Pattern 1: Template literal in SQL
async function getUserById_TemplateLiteral(userId: number) {
    const client = new Client();
    await client.connect();

    // VULNERABLE: SEC-001 should detect this
    const query = `SELECT * FROM users WHERE id = ${userId}`;
    const result = await client.query(query);

    await client.end();
    return result.rows;
}

// Pattern 2: String concatenation in SQL
async function getUserByUsername_Concatenation(username: string) {
    const client = new Client();
    await client.connect();

    // VULNERABLE: SEC-001 should detect this
    const query = "SELECT * FROM users WHERE username = '" + username + "'";
    const result = await client.query(query);

    await client.end();
    return result.rows;
}

// Additional patterns to be added in Phase 03

File: tests/fixtures/security/csharp/SqlInjectionVulnerable.cs
/*
 * Test Fixture: SQL Injection Vulnerable Patterns (C#)
 *
 * This file contains SQL injection vulnerability patterns for C# that MUST be detected.
 *
 * Expected detections: ≥3 violations
 * Rule ID: SEC-001
 * Severity: CRITICAL
 *
 * NOTE: Stub implementation - full patterns added during Phase 03 (Green)
 */

using System;
using System.Data.SqlClient;

namespace Security.Fixtures
{
    public class SqlInjectionVulnerable
    {
        // Pattern 1: String interpolation in SQL
        public void GetUserById_StringInterpolation(int userId)
        {
            // VULNERABLE: SEC-001 should detect this
            string query = $"SELECT * FROM Users WHERE Id = {userId}";

            using (var connection = new SqlConnection("..."))
            {
                var command = new SqlCommand(query, connection);
                connection.Open();
                command.ExecuteReader();
            }
        }

        // Pattern 2: String concatenation in SQL
        public void GetUserByUsername_Concatenation(string username)
        {
            // VULNERABLE: SEC-001 should detect this
            string query = "SELECT * FROM Users WHERE Username = '" + username + "'";

            using (var connection = new SqlConnection("..."))
            {
                var command = new SqlCommand(query, connection);
                connection.Open();
                command.ExecuteReader();
            }
        }

        // Additional patterns to be added in Phase 03
    }
}
